lib/ykpiv.c

```diff
@@ -1582,15 +1586,23 @@ static ykpiv_rc _ykpiv_verify(ykpiv_state *state, char *pin, size_t *p_pin_len,
       if (*p_pin_len < CB_PIN_MAX) {
         memset(apdu.st.data + *p_pin_len, 0xff, CB_PIN_MAX - *p_pin_len);
       }
-    }
-    else if (verify_spin && p_pin_len) {
+    } else if (verify_spin && p_pin_len) {
       apdu.st.data[0] = 0x01;
       apdu.st.data[1] = (uint8_t)*p_pin_len;
       memcpy(apdu.st.data + 2, pin, *p_pin_len);
-    }
-    else if (bio) {
+    } else if (bio) {
       memcpy(apdu.st.data, "\x02\x00", 2);
     }
+  } else {
+    if(bio) {
+      if (verify_spin) {
+        apdu.st.lc = 0;
+      } else {
+        memcpy(apdu.st.data, "\x03\x00", 2);
+      }
+    } else {
```

Comment:
This will return verification error without sending the blank verify for the user PIN.  This is used to check if PIV is selected (returns a non-PIN related error if PIV is not selected).